backend/src/db.rs

```diff
@@ -0,0 +1,32 @@
+//! Database related things.
+
+use anyhow::Result;
+use deadpool_postgres::{Config, Pool};
+use log::{debug, info, trace};
+use tokio_postgres::NoTls;
+
+use crate::config;
+
+
+/// Creates a new database connection pool.
+pub async fn create_pool(config: &config::Db) -> Result<Pool> {
+    let config = Config {
+        user: Some(config.user().into()),
+        password: Some(config.password().into()),
+        host: Some(config.host().into()),
+        port: Some(config.port),
+        dbname: Some(config.database.clone()),
+        .. Config::default()
+    };
+
+    trace!("Database configuration: {:#?}", config);
+
+    let pool = config.create_pool(NoTls)?;
+    info!("Created database pool");
+
+    // Test the connection by executing a simple query.
+    pool.get().await?.execute("SELECT 1", &[]).await?;
```

Comment:
Isn't this a waste? If it doesn't work, we will fail later anyway, but on the (hopefully more likely) happy path, this is just overhead. :wink:

Reply:
This is a tiny overhead on start up but makes the development process a *lot* less annoying. I think the backend should verify the DB connection works during startup. Sure, it would fail later, but it's nicer this way IMO. Also, overhead ... compared to all Java backends we could write, the startup of this one is a gazillion times faster anyway :D

Reply:
Sure, in the grand scheme of things the performance of this does not matter, and maybe it catches some mistakes early on. However, some of these errors should probably be handled explicitly at some later point. (For example the database not existing, which this apparently catches. Why does the connection call not catch this already?! Unfortunately that's outside our control for now. :shrug:)